addons/hr_timesheet/models/project.py

```diff
@@ -1,31 +1,99 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import models, fields, api
-from odoo.exceptions import UserError
-from odoo.tools.translate import _
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError, ValidationError
 
 
 class Project(models.Model):
     _inherit = "project.project"
 
     allow_timesheets = fields.Boolean("Allow timesheets", default=True)
+    analytic_account_id = fields.Many2one('account.analytic.account', string="Analytic Account", ondelete='set null',
+        help="Link this project to an analytic account if you need financial management on projects. "
+             "It enables you to connect projects with budgets, planning, cost and revenue analysis, timesheets on projects, etc.")
+
+    @api.onchange('partner_id')
+    def _onchange_partner_id(self):
+        domain = []
+        if self.partner_id:
+            domain = [('partner_id', '=', self.partner_id.id)]
+        return {'domain': {'analytic_account_id': domain}}
+
+    @api.constrains('allow_timesheets', 'analytic_account_id')
+    def _check_allow_timesheet(self):
+        for project in self:
+            if project.allow_timesheets and not project.analytic_account_id:
+                raise ValidationError(_('To allow timesheet, your project %s should have an analytic account set.' % (project.name,)))
+
+    @api.model
+    def name_create(self, name):
+        """ Create a project with name_create should generate analytic account creation """
+        values = {
+            'name': name,
+            'allow_timesheets': True,
+        }
+        return self.create(values).name_get()[0]
+
+    @api.model
+    def create(self, values):
+        """ Create an analytic account if project allow timesheet and don't provide one """
+        if values.get('allow_timesheets') and not values.get('analytic_account_id'):
```

Comment:
[Commit4] Don't forget to include my magnificient fix. Just throwing some flowers at myself.